modules/service-networking/variables.tf

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "project_id" {
+  description = "Project ID"
+  type        = string
+}
+
+variable "global_addresses" {
+  description = "List of global addresses to be created"
+  type = list(object({
+    name : string,
+    purpose : optional(string, "VPC_PEERING"),
+    type : optional(string, "INTERNAL"),
+    prefix_length : optional(number, 16)
+  }))
+}
+
+variable "network" {
+  description = "Network details including name and id"
+  type = object({
+    name = optional(string, null),
+    id   = string
+  })
```

Comment:
Why do we need both network name and id? Doesn't id of format `projects/{{project}}/global/networks/{{name}}` work in call cases?

Reply:
Made changes to use network name to derive network id